dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java

```diff
@@ -129,28 +133,39 @@ public static LinkageChecker create(Bom bom, Path exclusionFile)
     List<ClassPathEntry> artifactsInBom = classpath.subList(0, managedDependencies.size());
     ImmutableSet<ClassPathEntry> entryPoints = ImmutableSet.copyOf(artifactsInBom);
 
-    return LinkageChecker.create(classpath, entryPoints, exclusionFile);
+    return LinkageChecker.create(classpath, entryPoints, ImmutableList.of(), exclusionFile);
   }
 
   @VisibleForTesting
   LinkageChecker cloneWith(SymbolReferences newSymbolMaps) {
     return new LinkageChecker(
-        classDumper, classPath, newSymbolMaps, classReferenceGraph, excludedErrors);
+        classDumper, classPath, newSymbolMaps, classReferenceGraph, ImmutableList.of(), excludedErrors);
   }
 
   private LinkageChecker(
       ClassDumper classDumper,
       List<ClassPathEntry> classPath,
       SymbolReferences symbolReferenceMaps,
       ClassReferenceGraph classReferenceGraph,
+      List<Artifact> sourceFilterList,
       ExcludedErrors excludedErrors) {
     this.classDumper = Preconditions.checkNotNull(classDumper);
     this.classPath = ImmutableList.copyOf(classPath);
     this.classReferenceGraph = Preconditions.checkNotNull(classReferenceGraph);
     this.symbolReferences = Preconditions.checkNotNull(symbolReferenceMaps);
+    this.sourceFilterList = sourceFilterList;
     this.excludedErrors = Preconditions.checkNotNull(excludedErrors);
   }
 
+  /**
+   * Two artifacts are considered equal if the Maven Coordinates (GAV) are equal
+   */
+  private boolean areArtifactsEquals(Artifact artifact1, Artifact artifact2) {
```

Comment:
In source code comment, can you add why their equals method imolementations don't work?

Reply:
Added comments to the javadocs above.